File: rn/rnTestApp/data/firebase/services/firebaseRoleService.ts
import firestore from "@react-native-firebase/firestore";
import { UserRole } from "../../../domain/repository/AuthRepository";

export const firebaseRoleApi = {
    async getUserRole(userId: string): Promise<UserRole> {
    const snap = await firestore()
      .collection("korisnici")
      .doc(userId)
      .get();

    if (!snap.exists) {
      throw new Error("Korisnik ne postoji u bazi.");
    }

    const data = snap.data() ?? {};

    const jeSportas = (data as any).jeSportas === true;
    const jeTrener = (data as any).jeTrener === true;

    if (jeSportas) return "SPORTAS";
    if (jeTrener) return "TRENER";

    throw new Error("Uloga korisnika nije definirana.");
  },
};

File: kotlin_jetpack_compose/app/src/main/java/com/example/testapp/data/firebase/network/service/FirebaseRoleService.kt
package com.example.testapp.data.firebase.network.service

import com.example.testapp.domain.model.UserRole
import com.google.firebase.firestore.FirebaseFirestore
import kotlinx.coroutines.tasks.await

class FirebaseRoleService(
    private val firestore: FirebaseFirestore
) {
    suspend fun getUserRole(userId: String): UserRole {
        val snap = firestore.collection("korisnici").document(userId).get().await()
        if (!snap.exists()) {
            throw IllegalStateException("Korisnik ne postoji u bazi.")
        }

        val jeSportas = snap.getBoolean("jeSportas") ?: false
        val jeTrener = snap.getBoolean("jeTrener") ?: false

        return when {
            jeSportas -> UserRole.SPORTAS
            jeTrener -> UserRole.TRENER
            else -> throw IllegalStateException("Uloga korisnika nije definirana.")
        }
    }
}
